fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ElectionHandlerTest.java

```diff
@@ -199,4 +202,25 @@ public void testHandleElectionEnd() throws DataHandlingException {
     assertTrue(electionOpt.isPresent());
     assertEquals(EventState.CLOSED, electionOpt.get().getState());
   }
+
+  @Test
+  public void testHandleOpenElection() throws DataHandlingException {
+    OpenElection openElection = new OpenElection(lao.getId(), election.getId(), openedAt);
+    MessageGeneral message = new MessageGeneral(SENDER_KEY, openElection, GSON);
+
+    for (EventState state : EventState.values()) {
+      election.setEventState(state);
+
+      if (state == EventState.CREATED) {
+
+        //If the previous state was CREATED, it should change it to OPENED.
+        messageHandler.handleMessage(laoRepository, messageSender,
+            election.getChannel(), message);
+        assertEquals(EventState.OPENED, election.getState());
+        //Test for current TimeStamp
+        assertEquals(Instant.now().getEpochSecond(), election.getStartTimestamp());
```

Comment:
I am not sure, I understand. I think this should be equal to `openedAt`.

Reply:
When it opens, it should it modify the start time of the election (the current time). This is is the way it's implemented on the backend, so I made the same for java.

Reply:
This is an odd way to do it, and it might be a security issue.

Also, why does the open election data has an `opened_at` field then ?

I was not able to find the documentation on that specific message, but could you double-check with your team that it is the way it should be done ?